packages/stream-tile-handler/src/tile-document-handler.ts

```diff
@@ -210,8 +218,31 @@ export class TileDocumentHandler implements StreamHandler<TileDocument> {
   async _verifySignature(
     commitData: CommitData,
     context: Context,
-    controller: string
+    controller: string,
+    streamId?: StreamID
   ): Promise<void> {
+    if (streamId) {
+      const protectedHeaders = commitData.envelope.signatures.map((s) => s.protected)
+      const decodedProtectedHeaders = protectedHeaders.map((pH) => base64urlToJSON(pH))
+      const capHeaderIndex = decodedProtectedHeaders.findIndex((dph) => !!dph.cap)
```

Comment:
Hm, I think we should assume there's only one signature here. I believe we do so in `did.verifyJWS` anyway?

Reply:
Seems like it will inevitably lead to weird errors down the line...

Reply:
But how will you make sure that the CACAO object you verify is actually for the key that signed the particular protected header?

Reply:
why would it not be? tbh i havent seen a case where theres multiple signatures myself - could you elaborate on how that would happen?